sudo.go

```diff
@@ -4,18 +4,28 @@ import (
 	"errors"
 	"fmt"
 	"os"
-
-	"github.com/kardianos/osext"
 )
 
 const sudoArg = "__sudo"
 
+type SudoAction interface {
+	Name() string
+	Params() []string
+	Handle(params []string) error
+}
+
+var registeredActions map[string]SudoAction
+
+func RegisterAction(action SudoAction) {
+	registeredActions[action.Name()] = action
```

Comment:
I got a the following panic when testing this out:
```
panic: assignment to entry in nil map
```

Adding something like this to initialize the map at the top of `RegisterAction` should fix it:
```go
if registeredActions == nil {
    registeredActions = make(map[string]SudoAction)
}
```

Reply:
Oh, duh, thanks. I always forget that Go will let you use implicitly uninitialized data...

Reply:
Is there a reason to do

Reply:
Is there a reason to do
```go
var registeredActions map[string]SudoAction

func RegisterAction(action SudoAction) {
    if registeredActions == nil {
        registeredActions = make(map[string]SudoAction)
    }

    registeredActions[action.Name()] = action
}
```

instead of 

```go
var registeredActions = map[string]SudoAction{}

func RegisterAction(action SudoAction) {
    registeredActions[action.Name()] = action
}
```
?

Reply:
Either should work since we're initializing an empty map.  The former allows you to set a starting size for the map while the latter allows you to set initial values.  Based on [this](https://stackoverflow.com/a/31064737/11221577) it looks like it really just comes down to preference and how the map needs to be initialized.